cuda_core/cuda/core/experimental/_memory.pyx

```diff
@@ -558,6 +533,12 @@ class DeviceMemoryResourceAttributes:
     del mempool_property
 
 
+# Holds DeviceMemoryResource objects imported by this process.
+# This enables buffer serialization, as buffers can reduce to a pair
+# of comprising the memory resource UUID (the key into this registry)
+# and the serialized buffer descriptor.
+_ipc_registry = {}
```

Comment:
It seems there is no place where we'd purge items from this dict. I think it's an overlook? When the imported buffer is closed or goes out of scope, we should be able to do an inverse lookup and remove its MR from here, no? 

A growing internal cache could have potential performance impact.